cocos/particle/line.ts

```diff
@@ -119,152 +117,147 @@ export class Line extends Component {
 
     set positions (val) {
         this._positions = val;
-        if (this._model) {
-            this._model.addLineVertexData(this._positions, this._width, this._color);
+        if (this._models[0]) {
+            const lineModel = this._models[0] as LineModel;
+            lineModel.addLineVertexData(this._positions, this.width, this.color);
         }
     }
 
-    @type(CurveRange)
-    private _width = new CurveRange();
-
     /**
      * @zh 线段的宽度。
      */
+    @serializable
     @type(CurveRange)
     @range([0, 1])
     @displayOrder(3)
     @tooltip('i18n:line.width')
-    get width () {
-        return this._width;
-    }
+    public width = new CurveRange();
 
-    set width (val) {
-        this._width = val;
-        if (this._model) {
-            this._model.addLineVertexData(this._positions, this._width, this._color);
-        }
-    }
+    /**
+     * @zh 线段颜色。
+     */
+    @serializable
+    @type(GradientRange)
+    @displayOrder(6)
+    @tooltip('i18n:line.color')
+    public color = new GradientRange();
 
     @serializable
     private _tile = new Vec2(1, 1);
 
+    private _tile_offset: Vec4 = new Vec4();
+
     /**
      * @zh 图块数。
+     * @deprecated since v3.7
```

Comment:
Please describe the alternate in current version

Reply:
modified